autopilot/autopilot.go

```diff
@@ -84,55 +91,51 @@ func (ap *Autopilot) SetConfig(c Config) error {
 	return ap.store.SetConfig(c)
 }
 
-// Actions returns the autopilot actions that have occurred since the given time.
-func (ap *Autopilot) Actions(since time.Time, max int) []Action {
-	panic("unimplemented")
-}
-
 func (ap *Autopilot) Run() error {
+	// init ticker
+	if ap.ticker != nil {
+		return errors.New("autopilot is already running")
+	}
+	ap.ticker = time.NewTicker(ap.tickerDur)
+
+	// load config
 	if err := ap.load(); err != nil {
 		return err
 	}
 
-	go ap.contractLoop()
-	go ap.hostScanLoop()
-	<-ap.stopChan
-	return nil // TODO
+	// autopilot loop
+	for {
+		select {
+		case <-ap.stopChan:
+			return nil
+		case <-ap.ticker.C:
+		}
+
+		go ap.s.tryPerformHostScan()
+		_ = ap.c.performContractMaintenance() // TODO: handle error
```

Comment:
This is mostly just an example of how the contract maintenance loop can be blocking, while the host scan is launched in a goroutine. I personally like the tryPerformX thing which is a no-op if the subsystem decides it should not run for whatever reason (e.g. like a scan already running or not enough time passing). Open for feedback though